File: frontend/lib/models/user_profile_dto.dart
class UserProfileDTO {
  final String id;
  final String email;
  final String name;
  final String phoneNumber;
  final String? profilePicture;
  final double? latitude;
  final double? longitude;

  UserProfileDTO({
    required this.id,
    required this.email,
    required this.name,
    required this.phoneNumber,
    this.profilePicture,
    this.latitude,
    this.longitude,
  });

  factory UserProfileDTO.fromJson(Map<String, dynamic> json) {
    return UserProfileDTO(
      id: json['id'] as String,
      email: json['email'] as String,
      name: json['name'] as String,
      phoneNumber: json['phoneNumber'] as String,
      profilePicture: json['profilePicture'] as String?,
      latitude: json['latitude'] as double?,
      longitude: json['longitude'] as double?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'email': email,
      'name': name,
      'phoneNumber': phoneNumber,
      'profilePicture': profilePicture,
      'latitude': latitude,
      'longitude': longitude,
    };
  }
} 

File: backend/src/main/java/com/help/community/user/dto/UserProfileDTO.java
package com.help.community.user.dto;

import lombok.Builder;
import lombok.Data;

import java.util.Set;

/**
 * DTO para exponer el perfil del usuario autenticado.
 * Incluye información básica y conteo de solicitudes creadas.
 */
@Data
@Builder
public class UserProfileDTO {
    private Long id;
    private String email;
    private String name;
    private String phoneNumber;
    private Set<String> roles;
    private int createdRequestsCount;
    private Double latitude;
    private Double longitude;
}
